frontend-react/src/pages/admin/AdminMessageTestingPage/AdminMessageTestingPage.tsx

```diff
@@ -0,0 +1,132 @@
+import { GridContainer } from "@trussworks/react-uswds";
+import { useState } from "react";
+import { Helmet } from "react-helmet-async";
+import { useParams } from "react-router";
+import { AdminFormWrapper } from "../../../components/Admin/AdminFormWrapper";
+import { EditReceiverSettingsParams } from "../../../components/Admin/EditReceiverSettings";
+import MessageTestingForm from "../../../components/Admin/MessageTesting/MessageTestingForm";
+import MessageTestingResult from "../../../components/Admin/MessageTesting/MessageTestingResult";
+import Crumbs, { CrumbsProps } from "../../../components/Crumbs";
+import Spinner from "../../../components/Spinner";
+import { StaticAlert, StaticAlertType } from "../../../components/StaticAlert";
+import Title from "../../../components/Title";
+import { RSMessage } from "../../../config/endpoints/reports";
+import useTestMessageResult from "../../../hooks/api/messages/UseTestMessageResult/UseTestMessageResult";
+import useTestMessages from "../../../hooks/api/messages/UseTestMessages/UseTestMessages";
+import { FeatureName } from "../../../utils/FeatureName";
+
+export interface MessageTestingFormValuesInternal {
+    testMessage?: string;
+    testMessageBody?: string;
+}
+
+export interface MessageTestingFormValues {
+    testMessage: string;
+    testMessageBody: string;
+}
+
+enum MessageTestingSteps {
+    StepOne = "MessageTestSelection",
+    StepTwo = "MessageTestResults",
+}
+
+const AdminMessageTestingPage = () => {
+    const { orgname, receivername } = useParams<EditReceiverSettingsParams>();
+    const crumbProps: CrumbsProps = {
+        crumbList: [
+            {
+                label: FeatureName.RECEIVER_SETTINGS,
+                path: `/admin/orgreceiversettings/org/${orgname}/receiver/${receivername}/action/edit`,
+            },
+            { label: FeatureName.MESSAGE_TESTING },
+        ],
+    };
+
+    // Sets step of the Message Testing process laid out in the MessageTestingSteps enum
+    const [currentMessageTestStep, setCurrentMessageTestStep] = useState<MessageTestingSteps>(
+        MessageTestingSteps.StepOne,
+    );
+
+    // Sets data required for the MessageTestingForm
+    const { data: messageData, isDisabled } = useTestMessages();
+    const { setRequestBody, isLoading, data: testResultData } = useTestMessageResult();
+    const [selectedOption, setSelectedOption] = useState<RSMessage | null>(null);
+    const [currentTestMessages, setCurrentTestMessages] = useState<RSMessage[]>(messageData);
+    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        setRequestBody(selectedOption?.reportBody ? selectedOption.reportBody : null);
+        setCurrentMessageTestStep(MessageTestingSteps.StepTwo);
+    };
+
+    // If a user is _not_ a PrimeAdmin, block this feature
+    if (isDisabled) {
```

Comment:
We have a component wrapper `RequireGate` that can be used around this at the Router file level rather than doing it inside a page component. Though, if this is properly inserted inside the admin tree of the route object the check is likely already happening for us.

Reply:
Ah yes, this is redundant since page sits under:

```
path: "admin",
                element: (
                    <RequireGate auth={PERMISSIONS.PRIME_ADMIN}>
                        <Outlet />
                    </RequireGate>
                ),
```

as a child component. Will remove.